### PR TITLE
Oc 221 announcements

### PR DESCRIPTION
**Pull request by ZaraGlynn**

### CODE REVIEW

File: serverless.yml [hunk 1]
```diff
@@ -94,7 +95,8 @@ provider:
           - Ref: AWS::Region
           - Ref: AWS::AccountId
           - function:${self:service}-${opt:stage, self:provider.stage}-*
-
+ANNOUNCEMENTS_TABLE=tba21:
```
Comment (Dan-Wood) [inline serverless.yml]: Whats this?

File: serverless.yml [hunk 2]
```diff
@@ -486,8 +489,60 @@ functions:
...
+          method: patch
+          cors: true
+          authorizer: aws_iam
+
+  adminUpdateAnnouncement:
+    handler: src/admin/announcements/announcements.updateById
+    description: Update an announcement
+    events:
+      - http:
+          path: admin/announcements/updateById
```
Comment (Dan-Wood) [inline serverless.yml]: Just update? Can we update "by" anything else other than ID?

File: serverless.yml [hunk 3]
```diff
@@ -486,8 +489,60 @@ functions:
...
+    description: Insert an announcements
+    events:
+      - http:
+          path: admin/announcements
+          method: put
+          cors: true
+          authorizer: aws_iam
+
+  adminDeleteAnnouncement:
+    handler: src/admin/announcements/announcements.deleteById
```
Comment (Dan-Wood) [inline serverless.yml]: Just delete?

File: serverless.yml [hunk 4]
```diff
@@ -486,8 +489,60 @@ functions:
...
+    description: Removes an announcements
+    events:
+      - http:
+          path: admin/announcements
+          method: delete
+          cors: true
+          authorizer: aws_iam
+
+  adminGetAnnouncement:
+    handler: src/admin/announcements/announcements.get
```
Comment (Dan-Wood) [inline serverless.yml]: This isn't getByID , would be consistent if it was.

File: serverless.yml [hunk 5]
```diff
@@ -486,8 +489,60 @@ functions:
...
+    description: Get an announcements
+    events:
+      - http:
+          path: admin/announcements
+          method: get
+          cors: true
+          authorizer: aws_iam
+
```
Comment (Dan-Wood) [inline serverless.yml]: No caching?

File: src/admin/announcements/announcements.ts [hunk 6]
```diff
@@ -209,9 +208,9 @@ export const get = async(event: APIGatewayProxyEvent): Promise<APIGatewayProxyRe
           AND status = true
       `;
 
-    return successResponse({short_paths: await db.any(sqlStatement, params) });
+    return successResponse({announcement: await db.any(sqlStatement, params) });
```
Comment (Dan-Wood) [inline src/admin/announcements/announcements.ts]: This is get, with a required ID, but you're returning an array.

File: serverless.yml (hunk 1 above)
Comment (ZaraGlynn) [reply serverless.yml]: An accidental pasta

**Comment by Dan-Wood** [comment]
some extra comments

ChangeStatus could just be the update function, no need for another function to do what we already can do.

GET is admin only, should be admin and everyone else, along with contributor which can only see their own.

**Pull request merged**